src/main/java/software/amazon/smithy/lsp/ext/DependencyDownloader.java

```diff
@@ -0,0 +1,82 @@
+package software.amazon.smithy.lsp.ext;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+
+import coursierapi.Dependency;
+import coursierapi.Fetch;
+import coursierapi.MavenRepository;
+import coursierapi.Repository;
+import coursierapi.error.CoursierError;
+
+public class DependencyDownloader {
+  private List<Repository> repositories;
+  private List<Dependency> artifacts;
+  private Fetch fetch;
+
+  private static List<Repository> toRepositories(List<String> repos) {
+    // TODO: We assume maven repos by default.
+    return repos.stream().map(r -> MavenRepository.of(r)).collect(Collectors.toList());
+  }
+
+  private static List<Dependency> toDependencies(List<String> deps) throws ValidationException {
+    List<Dependency> lst = new LinkedList<Dependency>();
+
+    for (String dep : deps) {
+      // TODO: we assume Maven by default
+      String[] parts = dep.split(":");
+      if (parts.length != 3) {
```

Comment:
there is no Scala version per se, so it will just be a limitation that artifact must be fully qualified.